auth-jwt-service/src/main/java/io/stargate/auth/jwt/AuthzJwtService.java

```diff
@@ -60,6 +66,32 @@ public ResultSet authorizedDataRead(
 
     ResultSet result = action.call();
 
+    return filterUnauthorizedRows(result, stargateClaims);
+  }
+
+  @Override
+  public CompletionStage<ResultSet> authorizedAsyncDataRead(
+      Callable<CompletionStage<ResultSet>> action,
+      String token,
+      String keyspace,
+      String table,
+      List<TypedKeyValue> typedKeyValues,
+      SourceAPI sourceAPI) {
+    JSONObject stargateClaims;
+    CompletionStage<ResultSet> resultSetFuture;
+    try {
+      stargateClaims = extractClaimsFromJWT(token);
```

Comment:
from what I see, those two actions will be blocking, right?
Can't we make them async as well and chain using `CompletableFuture API`?
i.e.
```
CompletableFuture.supplyAsync(extractClaimsFromJWT).thenApply(preCheckDataReadWrite).thenCompose(action.call())`
```

Reply:
`extractClaimsFromJWT` should not do any I/O, AFAIK, it is supposed to just parse the token.

Reply:
still, even. if this is the only CPU bound the async method should be fully async, and not block

Reply:
Yeah parsing the token is CPU-bound and shouldn't be too intensive, so I don't see a strong incentive to do it asynchronously, it would just move the same work to a different executor.